compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2734,10 +2734,16 @@ private void assignTypesToMemberPatterns(BLangMatchPattern matchPattern, BType b
                     ((BArrayType) restMatchPattern.getBType()).eType = restType;
                     return;
                 }
-                if (patternType.tag != TypeTags.TUPLE) {
+                BTupleType patternTupleType;
+                if (patternType.tag == TypeTags.TUPLE) {
+                    patternTupleType = (BTupleType) patternType;
+                } else if (patternType.tag == TypeTags.INTERSECTION &&
```

Comment:
Wouldn't the same issue exist with mapping match patterns? Shouldn't we do something like

```java
        if (patternType.tag == TypeTags.INTERSECTION) {
            patternType = ((BIntersectionType) patternType).effectiveType;
        }
```

before the switch case?

Reply:
Yeah. Will update. Thanks.